### PR TITLE
update aws-sdk to sso session supported version

### PR DESCRIPTION
**Pull request by TheFynx**
# Description

Allows for the use of sso-session backed AWS Profiles

- Log out of cloud
 ```bash
❯ ~/.pulumi-dev/bin/pulumi logout
Logged out of https://app.pulumi.com
```

- Try to log in with awssdk=v2 with sso-session profile, existing error
 ```bash
❯ ~/.pulumi-dev/bin/pulumi login 's3://state-bucket?region=us-east-1&awssdk=v2&profile=AwsProfile'
error: problem logging in: unable to open bucket s3://state-bucket?region=us-east-1&awssdk=v2&profile=AwsProfile: open bucket s3://state-bucket?region=us-east-1&awssdk=v2&profile=AwsProfile: profile "AwsProfile" is configured to use SSO but is missing required configuration: sso_region, sso_start_url
```

- Try to log in with awssdk=v3 with sso-session profile, success
 ```bash
❯ ~/.pulumi-dev/bin/pulumi login 's3://state-bucket?region=us-east-1&awssdk=v3&profile=AwsProfile'
Logged in to levi-framework as levi (s3://state-bucket?region=us-east-1&awssdk=v3&profile=AwsProfile)
```

Fixes https://github.com/pulumi/pulumi/issues/13142

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`

- [ ] I have added tests that prove my fix is effective or that my feature works
	- Notes: Unsure if tests are needed to do a package update 

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: sdk/go/pulumi/config/require.go [hunk 1]
```diff
@@ -29,29 +29,21 @@ func failf(format string, a ...interface{}) {
 	panic(fmt.Errorf(format, a...))
 }
 
-func require(ctx *pulumi.Context, key string, secret bool, use, insteadOf string) string {
+func require(ctx *pulumi.Context, key, use, insteadOf string) string {
```
Comment (lukehoban) [inline sdk/go/pulumi/config/require.go:32]: These changes seem unrelated?  Should they be  backed out of this PR?

**Review by lukehoban — COMMENTED** [review]
Thanks for the PR. 

I believe the config changes are unrelated and unneeded for this PR?  They also seem to be causing test failures per https://github.com/pulumi/pulumi/actions/runs/5772945578/job/15648531383#step:34:734. 

Would love to separate those out so we can merge the dependency update.

File: sdk/go/pulumi/config/require.go (hunk 1 above)
Comment (TheFynx) [reply sdk/go/pulumi/config/require.go:32]: I have no idea how that got changed. I followed the contributing docs and that showed up changed after running the make commands. I included it since it said it was autogenerated. I can revert it to what's in the master branch

**Comment by Frassle** [comment]
Hmm not sure doesn't show anything as failed, lets hope something transient and try again

bors retry

**Comment by Frassle** [comment]
bors retry

**Comment by Frassle** [comment]
hmm ok bors seems really unhappy with this change. I'll see if anyone's got any ideas what's up, worst case we can try recreating this PR from one of our accounts and see if that behaves any better (can keep you tagged on the commit as author).

**Comment by TheFynx** [comment]
No worries, I'll just be happy if it gets merged in/updated one way or another.

**Comment by Frassle** [comment]
Looks like there may have been some github issues earlier, I'll give this another try

bors retry
